draft-ietf-jsonpath-base.md

```diff
@@ -541,6 +541,16 @@ The well-formedness and the validity of JSONPath queries are independent of
 the JSON value the query is applied to. No further errors relating to the
 well-formedness and the validity of a JSONPath query can be
 raised during application of the query to a value.
+This clearly separates well-formedness/validity errors in the query
+from mismatches that may actually stem from flaws in the data (see
+also {{data-attack}}).
+
+Often, mismatches between the structure expected by a valid query
+and the structure found in the data will lead to empty query results,
+which may be unexpected and indicate bugs in either.
+JSONPath implementations might therefore want to provide diagnostics
```

Comment:
suggest MAY?

Reply:
This is not an interoperability requirement.  It doesn't affect the other side.  We use MAY to say party A MAY do something, so party B ~~needs to~~ MUST be prepared for that behavior.  That's why I believe this is a "might" (non-2119 may).